File: java/LinkedList/src/LinkedList.java
/**
 * Created by Adarsh on 11/15/2015.
 */
public class LinkedList {

    Node head;

    class Node{
        int data;
        Node next;

        public Node(int d){
            data = d;
            next = null;
        }

        public void push(int d){

            Node new_node = new Node(d);
            new_node.next = head;
            head = new_node;

        }


        public void append(int d){
            Node new_node = new Node(d);

            if(head == null){
                head = new_node;
                return;
            }

            new_node.next = null;

            Node last = head;
            last = head;

            while(last.next != null){
                last = last.next;
            }

            last.next = new_node;
            return;
        }

        public void insertAfter(int prevData, int data){
            Node new_node = new Node(30);

            if(head == null){
                head = new_node;
                return;
            }

            Node prev = head;

            while(prev.data !=  prevData ){
                prev = prev.next;
            }

            new_node.next = prev.next;
            prev.next = new_node;

            return;


        }

    }
}

File: ruby/linkedList.rb
class Node
  attr_accessor :value,:next

  def initialize(value)
    @value = value
    @next = nil
  end
end

class LinkedList
  def initialize(val)
    @head = Node.new(val)
  end

  def addNode(val)
    current = @head
    while current.next != nil
      current = current.next
    end

    current.next = Node.new(val)
    self
  end

  def delete(val)
    current = @head
    if current.value == val
      @head = @head.next
    else
      while current.next.value != val && current.next != nil
        current = current.next
      end
      if (current != nil) && (current.next != nil)
        current.next = (current.next).next
      end
    end
  end


  def display
    current = @head
    while current.next !=nil
      print current.value," --> "
      current = current.next
    end
    print current.value
  end
end


ll = LinkedList.new(10)
ll.addNode(20)
ll.addNode(30)
ll.addNode(40)
ll.addNode(50)

ll.display